core-primitives/utils/src/buffer.rs

```diff
@@ -20,10 +20,12 @@
 use alloc::vec::Vec;
 
 /// Fills a given buffer with data and the left over buffer space with white spaces.
+/// Throw an error if the buffer size is not enough to hold `data`,
+/// return the length of `data` otherwise.
 pub fn write_slice_and_whitespace_pad(
```

Comment:
Can't we actually remove the padding entirely with this approach?

Reply:
I tried to drop the `right` part earlier, the problem was this fn was used in multiple places (more than just generating extrinsics) - it failed elsewhere, e.g. when getting pub key due to "trailing characters"
```
thread 'main' panicked at 'Invalid public key: Error("trailing characters", line: 1, column: 1418)', core-primitives/enclave-api/src/enclave_base.rs:315:59
```

Reply:
Ah, I see. Makes sense. 👍